lib/public/components/common/form/inputs/DateTimeInputModel.js

```diff
@@ -0,0 +1,316 @@
+/**
+ *  @license
+ *  Copyright CERN and copyright holders of ALICE O2. This software is
+ *  distributed under the terms of the GNU General Public License v3 (GPL
+ *  Version 3), copied verbatim in the file "COPYING".
+ *
+ *  See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ *  In applying this license CERN does not waive the privileges and immunities
+ *  granted to it by virtue of its status as an Intergovernmental Organization
+ *  or submit itself to any jurisdiction.
+ */
+
+import { extractUTCTimestampFromHTMLInput, formatUTCDateForHTMLInput } from '../../../../utilities/formatting/HTMLInputDateFormat.js';
+import { Observable } from '/js/src/index.js';
+import { MILLISECONDS_IN_ONE_DAY } from '../../../../utilities/dateUtils.js';
+
+/**
+ * @typedef DateTimeInputRawData
+ * @property {string} date the raw date value
+ * @property {string} time the raw time value
+ */
+
+/**
+ * @typedef DateTimeInputConfiguration
+ * @property {Partial<DateTimeInputRawData>} [defaults] the default raw values to use when partially updating inputs
+ * @property {boolean} [required] states if the input can be null
+ * @property {number} [min] the timestamp of the minimum date allowed by the input (included)
+ * @property {number} [max] the timestamp of the maximum date allowed by the input (excluded)
+ */
+
+/**
+ * Store the state of a date time model
+ */
+export class DateTimeInputModel extends Observable {
+    /**
+     * Constructor
+     * @param {DateTimeInputConfiguration} [configuration] the input configuration
+     */
+    constructor(configuration) {
+        super();
+        const { defaults = {}, required = false, min = null, max = null } = configuration;
+
+        /**
+         * @type {DateTimeInputRawData}
+         * @private
+         */
+        this._raw = { date: '', time: '' };
+
+        const { date: defaultDate = '', time: defaultTime = '' } = defaults;
+
+        /**
+         * @type {DateTimeInputRawData}
+         * @private
+         */
+        this._defaults = {
+            date: defaultDate,
+            time: defaultTime,
+        };
+
+        /**
+         * @type {number|undefined}
+         * @private
+         */
+        this._value = undefined;
+
+        this._visualChange$ = new Observable();
+
+        this._isRequired = required;
+        this._min = min;
+        this._max = max;
+    }
+
+    /**
+     * Returns the given date formatted in two parts, YYYY-MM-DD and HH:MM to fill in HTML date and time inputs
+     *
```

Comment:
Some docs starts from verb in third person (Returns) some not (Return)

Reply:
Returns is the right one (I suppose)

Reply:
yeah, I mentioned it just to keep it consistent